pkg/apis/serving/v1beta1/trained_model.go

```diff
@@ -0,0 +1,38 @@
+package v1beta1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type TrainedModel struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              TrainedModelSpec   `json:"spec,omitempty"`
+	Status            TrainedModelStatus `json:"status,omitempty"`
+}
+type TrainedModelSpec struct {
+	// Required field for parent inference service
+	InferenceService string `json:"inferenceService"`
+	// Predictor model spec
+	PredictorModel ModelSpec `json:"predictorModel"`
+	// Explainer model spec
+	ExplainerModel ModelSpec `json:"explainerModel,omitempty"`
+}
+type ModelSpec struct {
+	// Storage URI for the model repository
+	StorageURI string `json:"storageUri"`
+	// Default to latest
+	ModelVersionPolicy ModelVersionPolicy `json:"modelVersionPolicy,omitempty"`
+	// Machine Learning <framework name>-<git tag>
+	// The values could be: "tensorflow-v2.2.0","pytorch-v1.5.1","sklearn-0.23.1","onnx-v1.7.0","xgboost-v1.1.1", "myawesomeinternalframework-1.1.0" etc.
```

Comment:
Thoughts on making this a colon vs a dash to really reinforce the idea of a git tag?

Reply:
I think we can go with ":".  Hyphen is widely used in docker/git tags, it may confuse people. We need to implement a validation webhook to validate the format.